src/Http/Http.Extensions/test/RequestDelegateGenerator/Baselines/HandlesEndpointsWithAndWithoutDiagnostics.generated.txt

```diff
@@ -80,8 +81,11 @@ namespace Microsoft.AspNetCore.Http.Generated
                 var handler = Cast(del, global::System.String () => throw null!);
                 EndpointFilterDelegate? filteredInvocation = null;
                 var serviceProvider = options.ServiceProvider ?? options.EndpointBuilder.ApplicationServices;
-                var jsonOptions = serviceProvider?.GetService<IOptions<JsonOptions>>()?.Value ?? new JsonOptions();
-                var objectJsonTypeInfo = (JsonTypeInfo<object?>)jsonOptions.SerializerOptions.GetTypeInfo(typeof(object));
+                var jsonOptions = serviceProvider?.GetService<IOptions<JsonOptions>>()?.Value ?? FallbackJsonOptions;
+                var jsonSerializerOptions = jsonOptions.SerializerOptions;
+                jsonSerializerOptions.TypeInfoResolver ??= JsonTypeInfoResolver.Combine();
```

Comment:
Assuming the singleton is fully configured, couldn't this be written without any mutations as follows?
```suggestion
                var jsonSerializerOptions = serviceProvider?.GetService<IOptions<JsonOptions>>()?.Value?.SerializerOptions;
                if (jsonSerializerOptions is null or { TypeInfoResolver: null })
                {
                     // FallbackJsonOptions configured with the empty resolver.
                     jsonSerializerOptions = FallbackJsonOptions.Value.SerializerOptions;
                }
```

Reply:
Note that `JsonOptions FallbackJsonOptions = new();` does NOT have an empty resolver when `JsonSerializer.IsReflectionEnabledByDefault == false`.

Reply:
That would be great...wouldn't we run the risk of overriding scenarios where the user set the serializer options without an explicit type info resolver but other properties set?

```
builder.Services.ConfigureHttpJsonOptions(o => o.SerializerOptions.AllowTrailingCommas = true);
```

Reply:
> Note that JsonOptions FallbackJsonOptions = new(); does NOT have an empty resolver when JsonSerializer.IsReflectionEnabledByDefault == false.

Correct! One implication that this has for our tests is the fact that our RDG tests are implicitly using the reflection-based type resolver for scenarios where we don't define `JsonOptions`. I realized this after trying to initialize a JsonSerializerOptions from generated code and discovering the implications of _not_ using the one defined in `JsonOptions`.

Reply:
> Note that `JsonOptions FallbackJsonOptions = new();` does NOT have an empty resolver when `JsonSerializer.IsReflectionEnabledByDefault == false`.

That's true for `JsonOptions FallbackJsonOptions = new();`, but that's not necessarily true for `serviceProvider?.GetService<IOptions<JsonOptions>>()?.Value`.

Someone could call `JsonOptions.SerializerOptions.TypeInfoResolver = null`, but I see we've added a test for this. Now that we no longer assign to `JsonOptions.SerializerOptions.TypeInfoResolver` when its null like the code originally referenced in this comment, I think we're fine.